app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/BorrowedViewModel.java

```diff
@@ -1,22 +1,96 @@
 package com.cmput301f20t21.bookfriends.ui.library;
 
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
 import androidx.lifecycle.ViewModel;
 
+import com.cmput301f20t21.bookfriends.callbacks.OnSuccessCallbackWithMessage;
 import com.cmput301f20t21.bookfriends.entities.Book;
+import com.cmput301f20t21.bookfriends.enums.BOOK_STATUS;
+import com.cmput301f20t21.bookfriends.services.AuthService;
+import com.cmput301f20t21.bookfriends.services.BookService;
+import com.cmput301f20t21.bookfriends.services.RequestService;
+import com.google.firebase.firestore.DocumentSnapshot;
 
 import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
 
 public class BorrowedViewModel extends ViewModel {
-    public ArrayList<Book> getBooks() {
-        ArrayList<Book> mData = new ArrayList<>();
-        mData.add(new Book ("Some Book Name"));
-        mData.add(new Book ("How to Squish a Cat"));
-        mData.add(new Book ("Calculus II"));
-        mData.add(new Book ("Calculus III"));
-        mData.add(new Book ("Calculus IV"));
-        mData.add(new Book ("Calculator V"));
-        mData.add(new Book ("Terminator V"));
-        return mData;
+    private RequestService requestService = RequestService.getInstance();
+    private BookService bookService = BookService.getInstance();
+    private AuthService authService = AuthService.getInstance();
+
+    private MutableLiveData<List<Book>> books;
+    private MutableLiveData<Integer> updatedPosition;
```

Comment:
on @zhpeng811 's comment about adapter not initialized, that's a result of the book list mutable live data not initialized. we can init them on view model creation, i.e. 
```java
class BorrowedVM {
    private final MutableLiveData<...> = new MutableLiveData<>(new ArrayList<>());
}

```
(by final we make sure it's not re-initialized elsewhere via `= new MutableLiveData(...);` or we lost the reactivity )

just an option to counter the issue